File: TimeTracker.Server/GraphQL/Modules/Statistic/StatisticQueries.cs
using GraphQL;
using GraphQL.Types;
using Microsoft.Net.Http.Headers;
using TimeTracker.Business.Models;
using TimeTracker.Business.Repositories;
using TimeTracker.Server.Extensions;
using TimeTracker.Server.GraphQL.Modules.Auth;
using TimeTracker.Server.Services;

namespace TimeTracker.Server.GraphQL.Modules.Statistic
{
    public class StatisticQueries : ObjectGraphType
    {
        public StatisticQueries(IExcelExportRepository excelExportRepository, IUserRepository userRepository, CalendarDaysService calendarService)
        {
            Field<StatisticType, StatisticModel>()
                .Name("GetUserStatistic")
                .Argument<NonNullGraphType<GuidGraphType>, Guid>("UserId", "User id")
                .Argument<NonNullGraphType<DateTimeGraphType>, DateTime>("Date", "year and month for tracks")
                .ResolveAsync(async context =>
                {
                    ExcelModel exelModel;
                    var userId = context.GetArgument<Guid>("UserId");
                    var date = context.GetArgument<DateTime>("Date");
                    var user = await userRepository.GetByIdAsync(userId);
                    exelModel = user.ToExcelModel();
                    await excelExportRepository.GetUserHours(userId, date, exelModel);
                    exelModel.MonthHours = await calendarService.GetAmountWorkingHoursInMonth(date);
                    exelModel.PersentOfWork = exelModel.WorkerHours / exelModel.MonthHours;
                    var statiscticModel = new StatisticModel()
                    {
                        WorkerHours = exelModel.WorkerHours,
                        MonthHours = exelModel.MonthHours
                    };
                    return statiscticModel;
                }).AuthorizeWith(AuthPolicies.Authenticated);
        }
    }
}

File: TimeTracker.WebClient/src/modules/timeTracker/userStatistic/graphQL/statistic.queries.ts
import {gql} from '@apollo/client';
import {STATISTIC_FRAGMENT} from "./statistic.fragment";
import {Statistic} from "./statistic.type";

export type GetStatisticData = {statistic: {getUserStatistic: Statistic}}
export type GetStatisticInputType = { UserId: string, Date: string }

export const GET_USER_STATISTIC = gql`
    ${STATISTIC_FRAGMENT}
    query GetUserStatistic($UserId: Guid!, $Date:DateTime!){
        statistic{
            getUserStatistic(userId: $UserId, date: $Date){
                ...StatisticFragment
            }
        }
    }
`